lib/include/tick/robust/model_absolute_regression.h

```diff
@@ -8,25 +8,49 @@
 
 #include <cereal/types/base_class.hpp>
 
-class DLL_PUBLIC ModelAbsoluteRegression : public virtual ModelGeneralizedLinear {
+template <class T>
+class DLL_PUBLIC TModelAbsoluteRegression : public virtual TModelGeneralizedLinear<T> {
+ protected:
+  using TModelGeneralizedLinear<T>::features_norm_sq;
+  using TModelGeneralizedLinear<T>::compute_features_norm_sq;
+  using TModelGeneralizedLinear<T>::n_samples;
+  using TModelGeneralizedLinear<T>::n_features;
+  using TModelGeneralizedLinear<T>::fit_intercept;
+  using TModelGeneralizedLinear<T>::compute_grad_i;
+  using TModelGeneralizedLinear<T>::n_threads;
+  using TModelGeneralizedLinear<T>::get_inner_prod;
+
+ public:
+  using TModelGeneralizedLinear<T>::get_label;
+  using TModelGeneralizedLinear<T>::grad_i;
+  using TModelGeneralizedLinear<T>::get_features;
+  using TModelGeneralizedLinear<T>::grad_i_factor;
+
  public:
-  ModelAbsoluteRegression(const SBaseArrayDouble2dPtr features,
-                          const SArrayDoublePtr labels,
-                          const bool fit_intercept,
-                          const int n_threads = 1);
+  TModelAbsoluteRegression(
+    const std::shared_ptr<BaseArray2d<T> > features,
+    const std::shared_ptr<SArray<T> > labels,
```

Comment:
Is it somehow possible to have a shortcut for `std::shared_ptr<SArray<T> >` such as `SArrayPtr<T>`?

Reply:
A) Not without a macro
B) I don't think it will work in SWIG

Reply:
I think I actually tried it but SWIG was complaining about typeerrors

Reply:
Not cool, it might worth to open an issue to deal with this afterward. Not a big deal but it might simplify the code.